integration-tests/hibernate-orm-panache-kotlin/src/main/kotlin/io/quarkus/it/panache/kotlin/Dog.kt

```diff
@@ -0,0 +1,21 @@
+package io.quarkus.it.panache.kotlin
+
+import io.quarkus.hibernate.orm.panache.kotlin.PanacheCompanion
+import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntity
+import javax.persistence.Entity
+import javax.persistence.ManyToOne
+
+@Entity
+open class Dog() : PanacheEntity() {
+    companion object : PanacheCompanion<Dog>()
```

Comment:
Is there really no way to declare that in the `PanacheEntityBase` supertype? By making it self-parameterised with `PanacheEntity<Dog>`?

Reply:
companion objects aren't inherited so we can't go that route so far as I can tell.  Kotlin doesn't currently (and probably never will) support self types so I can't make that any less verbose.  At least, not that i've been able to determine.  It's not *bad*, though.  Just a touch redundant but it doesn't appear there's much to be done there.

Reply:
Ah right, they're not inherited. OK fine.